.travis.yml

```diff
@@ -17,10 +17,11 @@ jobs:
       script: |
         clang-format-7 --version
         # build a pathspec that excludes the files in .clang-format-ignore
-        while read file ; do echo EXCLUDES+="':(top,exclude)$file' " ; done < .clang-format-ignore
+        while read file ; do EXCLUDES+="':(top,exclude)$file' " ; done < .clang-format-ignore
         MERGE_BASE=$(git merge-base ${TRAVIS_COMMIT_RANGE%...*} ${TRAVIS_COMMIT_RANGE#*...})
         echo "Checking for formatting errors introduced since $MERGE_BASE"
-        git-clang-format-7 --binary clang-format-7 $MERGE_BASE -- $EXCLUDES
+        # use eval so that quotes (as inserted into $EXCLUDES above) are not interpreted as parts of file names
+        eval git-clang-format-7 --binary clang-format-7 $MERGE_BASE -- $EXCLUDES
```

Comment:
I know nested quoting gets really really messy so I may well be wrong, but should the eval line actually be something like:
`eval git-clang-format-7 --binary clang-format-7 "$MERGE_BASE" -- "$EXCLUDES"`
(the two varaiable expansions being in double quotes) ?

Reply:
Since in this case the `EXCLUDES` variable contains single-quotes that are being `eval`'d I think the double-quotes are unnecessary and perhaps wrong. The `MERGE_BASE` however ought to be quoted in line with usual practice

Reply:
`$MERGE_BASE` is not quoted just above and also wasn't quoted on this line before my changes.
I don't want to start adding quotes without testing that it still works and understanding why the quotes are necessary, if they are. Are you happy for me to merge this as it is now, or should I run all the tests with temp commits again with quotes?

Reply:
I haven't had time to test this with and without quotes for weeks. :disappointed: It worked when I tried it the way it is now (without quotes) so I'll merge this PR now. If it causes any problems then the quotes can easily be added in a follow-up PR.